src/Storages/ExternalStream/Kafka/WriteBufferFromKafkaSink.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <IO/BufferWithOwnMemory.h>
+#include <IO/WriteBuffer.h>
+#include <base/StringRef.h>
+
+namespace DB
+{
+
+class WriteBufferFromKafkaSink final : public BufferWithOwnMemory<WriteBuffer>
+{
+public:
+    explicit WriteBufferFromKafkaSink(const std::function<void(char * pos, size_t len)> callback, size_t buffer_size = DBMS_DEFAULT_BUFFER_SIZE)
```

Comment:
const object in parameter is almost always a bad idea since we have to copy everything to the target always and we probably have to copy everything to the parameter on the call site as well. We can do

```
explicit WriteBufferFromKafkaSink(std::function<...> on_next_, ...)
    : ..
    , on_next(std::move(on_next_))
{
    assert(on_next);
}
```